### PR TITLE
Add XPU ATen merge rule

### PR DESCRIPTION
**Pull request by EikanWang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122484

### CODE REVIEW

**Comment by EikanWang** [comment]
@atalman @huydhn , may I know if you have any comments on this PR?

**Review by huydhn — APPROVED** [review]

**Review by malfet — APPROVED** [review]

File: .github/merge_rules.yaml [hunk 1]
```diff
@@ -236,6 +236,21 @@
   - Lint
   - pull
 
+- name: XPU ATen
+  patterns:
+  - third_party/xpu.txt
```
Comment (malfet) [inline .github/merge_rules.yaml:241]: Don't you want to add more `c10/xpu`
```suggestion
  - aten/src/ATen/xpu/**
  - c10/xpu/**
  - third_party/xpu.txt
```

File: .github/merge_rules.yaml [hunk 2]
```diff
@@ -236,6 +236,21 @@
...
+  approved_by:
+  - EikanWang
+  - jgong5
+  - gujinghui
+  - alband
+  - malfet
+  - atalman
```
Comment (malfet) [inline .github/merge_rules.yaml:248]: I don't think this is needed here, as all 3 are already covered by different rules
```suggestion
```

**Pull request closed without merging**